layout/app/catalog/component.js

```diff
@@ -1,42 +1,90 @@
-import React, { PureComponent } from 'react';
-import PropTypes from 'prop-types';
+import React, { Component } from 'react';
 import debounce from 'lodash/debounce';
+import { toastr } from 'react-redux-toastr';
 
 // components
 import Layout from 'layout/layout/layout-app';
 import SearchInput from 'components/ui/SearchInput';
 import Spinner from 'components/ui/Spinner';
 import DatasetList from 'components/datasets/list';
+import Paginator from 'components/ui/Paginator';
+
+// services
+import { fetchDatasets } from 'services/dataset';
 
 // utils
 import { logEvent } from 'utils/analytics';
+import { DATASETS_PER_PAGE } from './constants';
+
+class CatalogLayout extends Component {
+  constructor(props) {
+    super(props);
 
-class CatalogLayout extends PureComponent {
-  static propTypes = {
-    list: PropTypes.array.isRequired,
-    loading: PropTypes.bool.isRequired,
-    getDatasets: PropTypes.func.isRequired,
-    setDatasetsSearch: PropTypes.func.isRequired
-  };
+    this.state = {
+      datasets: [],
+      loading: true,
+      search: null,
+      page: 1,
+      totalItems: 0
+    };
+
+    this.handlePageChange = this.handlePageChange.bind(this);
+  }
 
-  getSearchResults = debounce((value) => {
-    const { getDatasets } = this.props;
+  componentDidMount() {
```

Comment:
I think `componentWillMount` would be better, the sooner if triggers the fetch, the less the user has to wait for results.